projects/showcase/src/app/dynamic-router/dynamic-router.component.html

```diff
@@ -1,3 +1,175 @@
+            <ion-card>
+                <ion-card-header color="primary">
+                    <ion-card-subtitle>app-routing.module.ts</ion-card-subtitle>
+                </ion-card-header>
+                <ion-card-content>
+                    <pre>const routes: DynamicRoute&lt;NavigationTargets&gt;[] = [ ... ];</pre>
+                </ion-card-content>
+            </ion-card>
+        </li>
+        <li>For each non-nested route with a path defined, add a <code>navigationTarget</code> assignment to the corresponding enum value:
+            <ion-card>
+                <ion-card-header color="primary">
+                    <ion-card-subtitle>app-routing.module.ts</ion-card-subtitle>
+                </ion-card-header>
+                <ion-card-content>                    
+            <pre>
+const routes: DynamicRoute&lt;NavigationTargets&gt;[] = [
+    {{'{'}}
+        path: '',
+        redirectTo: 'home',
+        pathMatch: 'full'
+    },
+    {{'{'}}
+        path: 'home',
+        component: HomeComponent,
+        navigationTarget: NavigationTargets.Home
+    },
+    {{'{'}}
+        path: 'tasks',
+        component: TasksComponent,
+        navigationTarget: NavigationTargets.Tasks
+    },
+    ...
+];</pre>
+        </ion-card-content>
+    </ion-card>
+        </li>        
+    </ol>
+
+    <h3 id="nav-from-html">Navigation From HTML Templates <a routerLink="." fragment="nav-from-html">#</a></h3>
+    <ol>
+        <li>Add an enum property to the component to enable enum value binding:
+            <ion-card>
+                <ion-card-header color="primary">
+                    <ion-card-subtitle>app.component.ts</ion-card-subtitle>
+                </ion-card-header>
+                <ion-card-content>
+                    <pre>
+import {{'{'}} NavigationTargets } from './navigation-targets.enum';
+
+@Component({{'{'}}...})
+export class AppComponent {{'{'}}
+    ...
+
+    public NavigationTargets = NavigationTargets;
+}</pre>
+                </ion-card-content>
+            </ion-card>
+        </li>
+        <li>
+            Replace each <code>routerLink</code> directive and its hardcoded path with <code>iptDynamicRouterLink</code> and an enum value:
+            <ion-card>
+                <ion-card-header color="primary">
+                    <ion-card-subtitle>app.component.html</ion-card-subtitle>
+                </ion-card-header>
+                <ion-card-content>
+                    <pre>
+&lt;ion-item [button]="true" [iptDynamicRouterLink]="NavigationTargets.Tasks"&gt;
+    &lt;ion-label&gt;Tasks&lt;/ion-label&gt;
+&lt;/ion-item&gt;</pre>
+                </ion-card-content>
+            </ion-card>            
+        </li>
+    </ol>
+
+    <h3 id="nav-from-ts">Navigation From TypeScript Components <a routerLink="." fragment="nav-from-ts">#</a></h3>
+    <form (ngSubmit)="goToTarget()">
```

Comment:
This was supposed to be a demo where the user can pick a target and click a button to navigate but it doesn't work. The `goToTarget()` method doesn't get hit and it instead navigates to `http://localhost:4200/dynamic-router?ion-sel-0=1`

Reply:
Need to import `FormsModule` as well for this to work; ngSubmit is not doing anything right now